text/0032-upgrade-bootstrapper.md

```diff
@@ -0,0 +1,152 @@
+# `upgrade-bootstrapper`
+
+## Summary
+
+[summary]: #summary
+
+CEP PR: [casperlabs/ceps#0032](https://github.com/casperlabs/ceps/pull/0032)
+
+The `upgrade-bootstrapper` is a new binary which will be responsible for launching the `casper-node` binaries.  It will
+handle transitions between network versions by identifying when a current `casper-node` execution ceases and starting
+a new version of the `casper-node` binary.
+
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+The `upgrade-bootstrapper` will be a relatively simple Rust binary which will run the `casper-node` binary as a child
+process.  It will take a single optional argument: the path to the `config.toml` for the `casper-node` binary to be run.
+
+Under normal circumstances, the following steps will be performed:
+
+1. the `upgrade-bootstrapper` is started
+2. if the path to the node's `config.toml` was provided via an arg, the `upgrade-bootstrapper` writes the path value to
+   a hard-coded, known location (if writing fails, it panics).  Otherwise it reads the path value from the known
+   location.  If the path cannot be read from the known location, it panics
+3. the `upgrade-bootstrapper` spawns a child process, running the `casper-node` binary in validator mode, passing it the
+   path to its `config.toml` as an arg
+4. while they are running, before the network reaches the upgrade point, the next version of the `casper-node` gets
+   installed, but not run.  The binary is always named `casper-node-next` and its config named `config-next.toml`.  Each
+   gets installed to the same location as the current equivalent files.  The location of the `config.toml` can be
+   retrieved from the known location used by the `upgrade-bootstrapper` at step 2.  An upgrade file for chainspec
+   modifications should also be installed and its path included in the new `config-next.toml`
+5. when the current upgrade activation point is reached and the node has finished all its work in that era, it exits
```

Comment:
So it must be possible to inform a running node about a new activation point — is that already implemented/ticketed?

Reply:
Well, probably, but not necessarily.  I suggested three approaches in the first [unresolved question](https://github.com/CasperLabs/ceps/blob/91457cd963dc87a495e5b73c2ba92d159d2c9039/text/0032-upgrade-bootstrapper.md#unresolved-questions), but there could likely be a better way than these.

Reply:
Ah, right, I was thinking of the third approach. I think my vote goes to that one. :+1:

Reply:
There's no third approach though on that list.